Makefile

```diff
@@ -77,11 +77,16 @@ client: generate
 	hack/build-client.sh "${AUTOREST_IMAGE}" 2020-04-30 2021-09-01-preview 2022-04-01 2022-09-04 2023-04-01 2023-07-01-preview 2023-09-04 2023-11-22 2024-08-12-preview
 
 ci-rp: fix-macos-vendor
-	docker build . -f Dockerfile.ci-rp --ulimit=nofile=4096:4096 --build-arg REGISTRY=$(REGISTRY) --build-arg VERSION=$(VERSION) --no-cache=$(NO_CACHE)
+	docker build . -f Dockerfile.ci-rp --ulimit=nofile=4096:4096 --build-arg REGISTRY=$(REGISTRY) --build-arg VERSION=$(VERSION) --target builder --no-cache=$(NO_CACHE) -t aro-builder
```

Comment:
Is this step needed? With this line, it should build all the images in the multistage Dockerfile, correct?
```
docker build . -f Dockerfile.ci-rp --ulimit=nofile=4096:4096 --build-arg REGISTRY=$(REGISTRY) --build-arg VERSION=$(VERSION) -t aro-final
```

Reply:
I specifically asked for this, which is maybe a bit out of scope of this individual PR, but I was thinking a bit ahead to the point where we are building this dockerfile in PR builds on ephemeral CI agents without a docker cache. If we want to hit the cache from the previous build, we'll need to pull last build's `builder` image to save time, and to do that, we need to build/tag/push that stage individually.

Reply:
Ack, agreed on the comment to state the purpose.

Reply:
after a review, and with Shubhada's recent changes merged (which involved a scope increase) I talked with Mohammed and we decided to pull this change out - my bad, poor planning on my part.

I groomed additional work for this: https://issues.redhat.com/browse/ARO-7789

Reply:
I've done the roll back operation. And based on our discussion from last week I've committed my changes will rebase and try to resolve the merge conflicts first. And after that this PR will be ready for review.

Reply:
Regarding this https://issues.redhat.com/browse/ARO-7789 me and Brendan had discussion and we will track it with another PR and after rebase there will be some additional work to resolve merge conflicts as we got rid of ci-portal and consolidated all of those under one file which is ci-rp. And I'll take care of that.